src/Http/Http/src/StreamPipeReader.cs

```diff
@@ -309,7 +344,7 @@ private void AllocateReadTail()
                 _readHead.SetMemory(_pool.Rent(GetSegmentSize()));
                 _readTail = _readHead;
             }
-            else if (_readTail.WritableBytes == 0)
+            else if (_readTail.WritableBytes <= _minimumReadThreshold)
```

Comment:
```suggestion
            else if (_readTail.WritableBytes < _minimumReadThreshold)
```

Reply:
That should cause SetMinimumReadThresholdToMiminumSegmentSizeAlwaysGetsNewBlock to start failing which is good. The test should write at least one byte initially.